### PR TITLE
#1 Migration & Seeder DB

### PR DESCRIPTION
**Pull request by vinhhc97**
## Issue number
- #1

## The purpose
Init initial database

## Method of verification
1. Accessing docker container:
```
docker-compose exec php bash 
```

2. Execute a command - Running Migrations (https://laravel.com/docs/7.x/migrations#running-migrations): 
```
php artisan migrate
```

3. Running Seeders (https://laravel.com/docs/7.x/seeding#running-seeders):
```
php artisan db:seed
```

4. Check result in your Database Management UI:
image

## Others
https://docs.google.com/spreadsheets/d/12Z4jH4arqmLzo232QT618ao0Gc-n-aSPwBXnwTMW46o/edit#gid=0

### CODE REVIEW

File: backend/database/migrations/2023_07_19_081549_create_admins_table.php [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+class CreateAdminsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('admins', function (Blueprint $table) {
```
Comment (ckthanh-sgt) [inline backend/database/migrations/2023_07_19_081549_create_admins_table.php:16]: @vinhhc97 
Please create model for `Admin`, `Grade`, `Student` in the following path `app/Http/Models/...php`
image

Docs:
https://laravel.com/docs/7.x/eloquent#defining-models

File: backend/database/migrations/2023_07_19_081549_create_admins_table.php [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+            $table->id();
+            $table->string('username',20)->unique()->min(6);
+            $table->string('password',255);
+            $table->string('email',20)->unique();
```
Comment (ckthanh-sgt) [inline backend/database/migrations/2023_07_19_081549_create_admins_table.php:20]: @vinhhc97 
Refactor code

- How to do:
Press combination of `command` + `option` + `L` at the same time

File: backend/database/migrations/2023_07_19_082545_create_students_table.php [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('students', function (Blueprint $table) {
+            $table->id();
+            $table->string('name',20);
+            $table->date('birthday');
+        });
```
Comment (ckthanh-sgt) [inline backend/database/migrations/2023_07_19_082545_create_students_table.php:20]: @vinhhc97 
Refactor code for these lines

```
$table->id();
$table->string('name',20);
$table->date('birthday');
```

- How to do:
Press combination of `command` + `option` + `L` at the same time

File: backend/database/migrations/2023_07_19_082723_create_grades_table.php [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('grades', function (Blueprint $table) {
+            $table->id();
+            $table->foreignid('student_id')->constrained('student.id');
+            $table->enum('subject', ['Math', 'Music', 'English']);
+            $table->integer('grades');
```
Comment (ckthanh-sgt) [inline backend/database/migrations/2023_07_19_082723_create_grades_table.php:20]: @vinhhc97 
Refactor code for these lines:

```
$table->foreignid('student_id')->constrained('student.id');
$table->enum('subject', ['Math', 'Music', 'English']);
$table->integer('grades');
```

- How to do:
Press combination of `command` + `option` + `L` at the same time

File: backend/database/migrations/2023_07_19_082723_create_grades_table.php [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    /**
+     * Run the migrations.
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::create('grades', function (Blueprint $table) {
+            $table->id();
+            $table->foreignid('student_id')->constrained('student.id');
```
Comment (ckthanh-sgt) [inline backend/database/migrations/2023_07_19_082723_create_grades_table.php:18]: @vinhhc97 
Wrong method for `$table->foreignid`

- Doc:
https://laravel.com/docs/7.x/migrations#creating-columns

image

File: backend/database/seeds/AdminSeeder.php [hunk 6]
```diff
@@ -0,0 +1,20 @@
...
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+        DB::table('admins')->insert([
+            'username'=>'vinh',
+            'password'=>'123bonnamsau@@',
+            'email'=>'vinh.hc@gmail.com'
```
Comment (ckthanh-sgt) [inline backend/database/seeds/AdminSeeder.php:17]: @vinhhc97 
Refactor code for these lines

```
'username'=>'vinh',
'password'=>'123bonnamsau@@',
'email'=>'vinh.hc@gmail.com'
```

- How to do:
Press combination of `command` + `option` + `L` at the same time

File: backend/database/seeds/DatabaseSeeder.php [hunk 7]
```diff
@@ -12,5 +12,8 @@ class DatabaseSeeder extends Seeder
     public function run()
     {
         // $this->call(UserSeeder::class);
```
Comment (ckthanh-sgt) [inline backend/database/seeds/DatabaseSeeder.php:14]: @vinhhc97 
Remove unused line

File: backend/database/seeds/GradeSeeder.php [hunk 8]
```diff
@@ -0,0 +1,20 @@
...
+     *
+     * @return void
+     */
+    public function run()
+    {
+        DB::table('grades')->insert([
+            'student_id'=>465,
+            'subject'=>'English',
+            'grade'=>9
+        ]);
```
Comment (ckthanh-sgt) [inline backend/database/seeds/GradeSeeder.php:18]: @vinhhc97 
Refactor code for these lines

- How to do:
Press combination of `command` + `option` + `L` at the same time

File: backend/database/seeds/StudentSeeder.php [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+     * Run the database seeds.
+     *
+     * @return void
+     */
+    public function run()
+    {
+       DB::table('Students')->insert([
+           'name'=>'VinhHoang',
+           'birthday'=>'09/11/1997'
+       ]);
```
Comment (ckthanh-sgt) [inline backend/database/seeds/StudentSeeder.php:17]: @vinhhc97 
Refactor code for these lines

- How to do:
Press combination of `command` + `option` + `L` at the same time

Comment (ckthanh-sgt) [inline backend/database/seeds/StudentSeeder.php:17]: @vinhhc97 
Please create `Factory` for `Student`, `Admin`, `Grades` to create more data

- Docs:
https://laravel.com/docs/7.x/database-testing#generating-factories

**Review by ckthanh-sgt — CHANGES_REQUESTED** [review]
@vinhhc97 
I've commented, please confirm

File: backend/database/seeds/GradeSeeder.php [hunk 10]
```diff
@@ -0,0 +1,10 @@
+<?php
+
+use Illuminate\Database\Seeder;
+
+class GradeSeeder extends Seeder
+{
+    public function run()
+    {
+    }
```
Comment (hdquangsgt) [inline backend/database/seeds/GradeSeeder.php:9]: @vinhhc97 
Please add record for GradeSeeder

File: backend/database/seeds/DatabaseSeeder.php [hunk 11]
```diff
@@ -11,6 +11,7 @@ class DatabaseSeeder extends Seeder
      */
     public function run()
     {
-        // $this->call(UserSeeder::class);
+        $this->call(AdminSeeder::class);
+        $this->call(StudentSeeder::class);
```
Comment (hdquangsgt) [inline backend/database/seeds/DatabaseSeeder.php:15]: @vinhhc97 
GradeSeeder???

**Review by hdquangsgt — COMMENTED** [review]
@vinhhc97 
Please check my comments

File: backend/app/Grade.php [hunk 12]
```diff
@@ -0,0 +1,18 @@
+<?php
+
+namespace App;
+
+use Illuminate\Database\Eloquent\Model;
+
+class Grade extends Model
+{
+    protected $fillable =
+        ['student_id', 'subject', 'grade'];
```
Comment (hdquangsgt) [inline backend/app/Grade.php:10]: @vinhhc97 
Please format code

File: backend/app/Admin.php [hunk 13]
```diff
@@ -0,0 +1,15 @@
...
+namespace App;
+use Illuminate\Foundation\Auth\User as Authenticatable;
+use Illuminate\Notifications\Notifiable;
+
+class Admin extends Authenticatable
+{
+    use Notifiable;
+
+    protected $fillable =
+        ['username', 'password', 'email',];
```
Comment (hdquangsgt) [inline backend/app/Admin.php:12]: @vinhhc97 
Please format code

File: backend/database/seeds/DatabaseSeeder.php [hunk 14]
```diff
@@ -11,6 +14,7 @@ class DatabaseSeeder extends Seeder
...
```
Comment (hdquangsgt) [inline backend/database/seeds/DatabaseSeeder.php:18]: @vinhhc97 
```suggestion
        $this->call([
            AdminSeeder::class, 
            StudentSeeder::class,
        ]);
```

**Review by hdquangsgt — COMMENTED** [review]
@vinhhc97 
Please confirm

**Review by hdquangsgt — APPROVED** [review]
@vinhhc97 
LGTM!

**Pull request merged**